lib/connectors/crawler/scheduler.rb

```diff
@@ -22,11 +22,50 @@ def connector_settings
         []
       end
 
+      def when_triggered
+        loop do
+          connector_settings.each do |cs|
+            # crawler only supports :sync
+            if sync_triggered?(cs)
+              yield cs, :sync, nil
+              next
+            end
+
+            custom_schedule_triggered, schedule_key = custom_schedule_triggered?(cs)
+            if custom_schedule_triggered
+              yield cs, :sync, schedule_key
```

Comment:
I also considered the following, but I wasn't sure if it was wise to change the structure of the connector settings:

```ruby
yield cs.merge(scheduling_key: scheduling_key), :sync
```

Reply:
I can't see the comment you are replying to but I totally agree that it's wise **not to change** the structure of the connector settings. We don't have a concept of "virtual" fields so it can lead to unexpected behavior

Reply:
I wasn't replying, just pre-emptively explaining 😄 
But good to know I was on the right track!